src/Content/Version.php

```diff
@@ -33,39 +35,46 @@ public function content(string $language = 'default'): Content
 	{
 		return new Content(
 			parent: $this->model,
-			data:   $this->model->storage()->read($this->id, $language),
+			data:   $this->model->storage()->read($this->id, $this->language($language)),
 		);
 	}
 
+	/**
+	 * Provides simplified access to the absolute content file path.
+	 * This should stay an internal method and be removed as soon as
+	 * the dependency on file storage methods is resolved more clearly.
+	 *
+	 * @internal
+	 */
+	public function contentFile(string $language = 'default'): string
+	{
+		return $this->model->storage()->contentFile($this->id, $this->language($language));
+	}
+
 	/**
 	 * Creates a new version for the given language
+	 *
+	 * @param array<string, string> $fields Content fields
 	 */
 	public function create(array $fields, string $language = 'default'): void
 	{
-		$this->model->storage()->create($this->id, $language, $fields);
+		$this->model->storage()->create($this->id, $this->language($language), $fields);
 	}
 
 	/**
-	 * Deletes a version by language or for any language
+	 * Deletes a version with all its languages
 	 */
-	public function delete(string|null $language = null): void
+	public function delete(): void
 	{
-		// delete all languages
-		if ($language === null) {
-			foreach ($this->model->kirby()->languages() as $language) {
-				$this->model->storage()->delete($this->id, $language->code());
-			}
-		}
-
 		// delete the default language in single-language mode
 		if ($this->model->kirby()->multilang() === false) {
-			$this->model->storage()->delete($this->id, 'default');
-			return;
+			$this->model->storage()->delete($this->id, $this->language('default'));
```

Comment:
`Suggestion`: The `return` statement doesn't do much, but it could improve clarity and also slightly increase performance in single-lang:

```suggestion
			$this->model->storage()->delete($this->id, $this->language('default'));
			return;
```

Reply:
This really looks like an oversight on my end. The return is definitely needed.